File: server/src/main/java/ru/vitaliy/petrov/server/forms/requests/documents/AdministrativeOffenceSceneSchemeUpdateRequest.java
package ru.vitaliy.petrov.server.forms.requests.documents;

import lombok.Data;

import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.NotNull;
import java.sql.Date;
import java.sql.Time;

@Data
public class AdministrativeOffenceSceneSchemeUpdateRequest {

    @NotNull
    private final Long documentID;

    @NotNull
    private final Date updatedDateOfFill;

    @NotNull
    private final Time updatedTimeOfFill;

    @NotNull
    @NotEmpty
    private final String updatedPlaceOfFill;

    @NotNull
    @NotEmpty
    private final String fileLink;

    @NotNull
    @NotEmpty
    private final String updatedSchemeConventions;

    @NotNull
    private final Long updatedFirstWitnessID;

    @NotNull
    private final Long updatedSecondWitnessID;

    @NotNull
    private final Long carAccidentEntityID;
}

File: 3.client/app/src/main/java/com/petrov/vitaliy/caraccidentapp/domain/models/requests/documents/AdministrativeOffenceSceneSchemeUpdateRequest.kt
package com.petrov.vitaliy.caraccidentapp.domain.models.requests.documents

import java.sql.Date
import java.sql.Time

data class AdministrativeOffenceSceneSchemeUpdateRequest(
    val documentID: Long,
    val updatedDateOfFill: Date,
    val updatedTimeOfFill: Time,
    val updatedPlaceOfFill: String,
    val fileLink: String,
    val updatedSchemeConventions: String,
    val updatedFirstWitnessID: Long,
    val updatedSecondWitnessID: Long,
    val carAccidentEntityID: Long
)
